### PR TITLE
lesson 6.1 

### PR DESCRIPTION
**Pull request by wessstt**
Lesson 6.1:
— Added projects section, nav menu, and image drop shadow to index.js
— Added more styles to index.css 
— Added projects and nav section to index.html

### CODE REVIEW

File: js/index.js [hunk 1]
```diff
@@ -1,24 +1,59 @@
...
+    })
+  );
+
+  /*
+     ========================================== 
+     ************ ABOUT SECTION  *************
+     ==========================================
+   */
+    const dropShadow = document.getElementById("img");
+    dropShadow.style.setProperty("-webkit-filter", "drop-shadow(12px 12px 10px #072b3d)");
```
Comment (mccalluc) [inline js/index.js]: I think this is unconditional? I think it'd be simpler to just apply it in CSS.

I'm also not sure if the `-webkit-` prefix is necessary. https://caniuse.com/?search=filter

File: js/index.js [hunk 2]
```diff
@@ -27,10 +62,40 @@ document.addEventListener("DOMContentLoaded", () => {
...
+    if (githubRequest.readyState === 4 && githubRequest.status === 200) {
+      let repositories = JSON.parse(this.response);
+      const projectSection = document.getElementById("projects");
+      const projectList = projectSection.querySelector("ul");
+
+      for (let i = 0; i < repositories.length; i++) {
+        const project = document.createElement("li");
+        // change list to buttons that link to github
+        const repoLinks = document.createElement("button");
+        repoLinks.href = repositories[i].html_url;
```
Comment (mccalluc) [inline js/index.js]: I don't think `<button>` supports `href`: You probably want to just make these `<a>` elements, and then style them as round rects.

File: js/index.js [hunk 3]
```diff
@@ -27,10 +62,40 @@ document.addEventListener("DOMContentLoaded", () => {
...
+  githubRequest.addEventListener("load", function () {
+    if (githubRequest.readyState === 4 && githubRequest.status === 200) {
+      let repositories = JSON.parse(this.response);
+      const projectSection = document.getElementById("projects");
+      const projectList = projectSection.querySelector("ul");
+
+      for (let i = 0; i < repositories.length; i++) {
+        const project = document.createElement("li");
+        // change list to buttons that link to github
+        const repoLinks = document.createElement("button");
```
Comment (mccalluc) [inline js/index.js]: Since it's only going to be one link, I'd use singular here.

File: js/index.js [hunk 4]
```diff
@@ -69,7 +134,7 @@ document.addEventListener("DOMContentLoaded", () => {
       const entry = e.target.parentNode;
       messageList.removeChild(entry);
       if (messageList.length === true) {
```
Comment (mccalluc) [inline js/index.js]: Does this work for you? I would never expect `length` to be `true`... but if it's non-zero, it's true-ish, and you could do this:
```suggestion
      if (messageList.length) {
```

**Review by mccalluc — COMMENTED** [review]
End result looks really good! There's a lot in here, and it would probably be good to check that a new feature works the way you expect before moving to the next. I noticed that the repo buttons don't work as links, and there's also a bug where clicking to remove one message actually hides them all.

File: js/index.js (hunk 4 above)
Comment (mccalluc) [reply js/index.js]: ... this might also be related to the message removal bug.

File: js/index.js (hunk 1 above)
Comment (wessstt) [reply js/index.js]: awesome, thanks. Yeah, I wasn't sure about it. I had it originally in css and then was having issues with it displaying over top of my menu so I tried it in JS to see if it helped and then realized it was the z-index haha. 

I went ahead and moved it back to css.

File: js/index.js (hunk 2 above)
Comment (wessstt) [reply js/index.js]: oops. Forget to check again after I switched to buttons. 

Fixed.

File: js/index.js (hunk 4 above)
Comment (wessstt) [reply js/index.js]: thanks for the note. Once I made the suggested change it wasn't hiding my header after removing all the messages. I realized I needed to call (messageList.children.length) in the if statement. Now it's working properly. 🙌

**Pull request merged**